network/server.go

```diff
@@ -523,6 +523,17 @@ func (s *Server) updateBootnodeConnCount(peerID peer.ID, delta int64) {
 	s.bootnodes.increaseBootnodeConnCount(delta)
 }
 
+// ShouldIgnoreSyncToPeer Check if client should ignore this peer when determining best peer to sync to
+func (s *Server) ShouldIgnoreSyncToPeer(peerID peer.ID) bool {
+	// If the peer is not a bootnode and the client has BootnodeOnlySync enabled
+	ignoreSync := s.config.BootnodeOnlySync && !s.bootnodes.isBootnode(peerID)
+	if ignoreSync {
+		s.logger.Debug("avoid syncing to non-bootnode peer", "id", peerID.String())
+	}
+
+	return ignoreSync
+}
```

Comment:
Right now only syncer syncs some data between peers. I think network layer shouldn't have such a function. Network layer just provides a method to return a peer is bootnode or not.